src/nanoemoji/util.py

```diff
@@ -85,30 +87,120 @@ def file_printer(filename):
             yield partial(print, file=f)
 
 
-def ensure_ttfont_fully_decompiled(ttfont: ttLib.TTFont):
-    # A TTFont might be opened lazily and some tables only partially decompiled.
-    # So for this to work on any TTFont, we first compile everything to a temporary
-    # stream then reload with lazy=False. Input font is modified in-place.
+def _assert_all_keys_loaded(font: ttLib.TTFont):
+    not_loaded = sorted(t for t in font.keys() if not font.isLoaded(t))
+    assert (
+        not not_loaded
+    ), f"Everything should be loaded, following aren't: {not_loaded}"
+
+
+def _reload(font: ttLib.TTFont, lazy: bool = True):
+    # Stream font to memory and load it back again
     tmp = BytesIO()
-    ttfont.save(tmp)
+    font.save(tmp)
     tmp.seek(0)
-    ttfont2 = ttLib.TTFont(tmp, lazy=False)
-    for tag in ttfont2.keys():
-        table = ttfont2[tag]
+    font2 = ttLib.TTFont(tmp, lazy=lazy)
+
+
+def load_fully(font: Union[Path, ttLib.TTFont]) -> ttLib.TTFont:
+    if isinstance(font, Path):
+        font = ttLib.TTFont(str(font), lazy=False)
+    else:
+        # A TTFont might be opened lazily and some tables only partially decompiled.
+        # If so, reload it
+        if font.lazy:
+            font = _reload(font, lazy=False)
+
+    # one would think lazy=False meant tags would report isLoaded true
+    # one would be wrong, following fails:
+    # _assert_all_keys_loaded(font)
+
+    # OK, for those of you that are STILL lazy just bloody load
+    for tag in font.keys():
+        table = font[tag]
         # cmap is exceptional in that it always loads subtables lazily upon getting
         # their attributes, no matter the value of TTFont.lazy option.
         # TODO: remove this hack once fixed in fonttools upstream
         if tag == "cmap":
             _ = [st.cmap for st in table.tables]
-        ttfont[tag] = table
 
+    _assert_all_keys_loaded(font)
+
+    # post has a cool pattern of raising errors if it has no glyphOrder attribute
+    # font isn't all that useful w/o glyph order so lets confirm we're ok on that front
+    # TODO contemplate whether this is simply a FontTools bug
+    font.getGlyphOrder()
+
+    return font
+
+
+class OtDataPathEntry(NamedTuple):
+    value: Any
+    attr_name: str  # attr of parent to get value
+
+
+OtDataPath = Tuple[OtDataPathEntry, ...]
+
+
+def traverse_ot_data(root, callback: Callable[[OtDataPath], None]) -> Iterable[Any]:
+    visited = set()
+    frontier: List[OtDataPath] = [(OtDataPathEntry(root, ""),)]
+    while frontier:
+        # path is (value, attr_name) tuples. attr_name is attr of parent to get value
+        path = frontier.pop(0)
+        current = path[-1].value
+        visited.add(id(current))
 
-def reorder_glyphs(ttfont: ttLib.TTFont, glyph_order: List[str]):
+        callback(path)
+
+        converters = getattr(current, "converters", [])
```

Comment:
BaseTable has a `getConverters` public method, you should use that instead of reaching directly to the converters yourself

Reply:
the reason you are getting garbage for the format-switching subtables is that accessing the inner `converters` attribute gives you a dict (keyed by format) not a list in those cases. You want to use `getConverters()` instead which returns the correct list of converters all the time for both normal and format-switching subtables

Reply:
Migrated to your shiny new FontTools iterSubTables